redesign_tests/test_subgenerators.py

```diff
@@ -1,23 +1,60 @@
 import unittest
 
-from redesign.generator_builder import Generator
-import pandas as pd
 import dataprofiler as dp
+import pandas as pd
+
+from redesign.generator_builder import Generator
 
-class TestSubgenerators(unittest.TestCase):
 
+class TestSubgenerators(unittest.TestCase):
     @classmethod
     def setUpClass(cls):
-        tab_data = dp.Data("redesign_tests/tabular.csv")
-        cls.tab_profile = dp.Profiler(tab_data)
+        profile_options = dp.ProfilerOptions()
+        profile_options.set(
+            {
+                "data_labeler.is_enabled": False,
+                "correlation.is_enabled": True,
+                "multiprocess.is_enabled": False,
+            }
+        )
 
-        unstruct_data = pd.Series(['first string', 'second string'])
-        cls.unstruct_profile = dp.Profiler(unstruct_data, profiler_type='unstructured')   
+        data = dp.Data("redesign_tests/iris.csv")
+        cls.tab_profile = dp.Profiler(
+            data, profiler_type="structured", options=profile_options
+        )
+
+        data = pd.Series(["first string", "second string"])
+        cls.unstruct_profile = dp.Profiler(
+            data, profiler_type="unstructured", options=profile_options
+        )
+
+        data = dp.Data("redesign_tests/graph.csv")
+        cls.graph_profile = dp.Profiler(
+            data, profiler_type="graph", options=profile_options
+        )
 
     def test_synthesize_tab(self):
-        result = Generator(self.tab_profile).synthesize()
-        self.assertEqual(result, "Synthesized tabular data!")
-    
+        result = Generator(profile=self.tab_profile).synthesize(100)
+        self.assertEqual(len(result), 100)
+
     def test_synthesize_unstruct(self):
-        result = Generator(self.unstruct_profile).synthesize()
-        self.assertEqual(result, "Synthesized unstructured data!")
+        with self.assertRaises(NotImplementedError):
+            Generator(profile=self.unstruct_profile).synthesize()
+
+    def test_synthesize_graph(self):
+        with self.assertRaises(NotImplementedError):
+            Generator(profile=self.graph_profile).synthesize()
+
+    def test_no_profile(self):
+        with self.assertRaises(
+            ValueError,
+            msg="No profile object was passed in kwargs. If you want to generate synthetic data from a profile, pass in a profile object through kwargs.",
```

Comment:
to be more explicit, I think this should specify the key under which the user should pass the profile object... i.e. the key being `profile` under which the profile object should be passed